### DISCUSSION TITLE
pull requests triggered from actions don't trigger other actions

### DISCUSSION THREAD

**Discussion by OmarTawfik** [community/community#57484] [Actions] [answered]
### Select Topic Area

Bug

### Body

I have  [the following workflow](https://github.com/NomicFoundation/slang/blob/3a1983fcabc8f893529ccb4a25b205916bf6fadc/.github/workflows/check.yml) that is triggered on all pull requests, and all pushes except to `main` branch:

```yml
on:
  workflow_dispatch: {}
  pull_request: {}
  push:
    branches-ignore:
      - "main"
```

It runs correctly when I push a branch from my laptop, or when a PR is created from a user fork.

Some actions can also push new branches (to the same repository) and create PRs for them:

- https://github.com/NomicFoundation/slang/pull/492
- https://github.com/NomicFoundation/slang/pull/493

However, these PRs don't trigger that workflow. I have to close and re-open the PR in order for the workflow to be triggered correctly.

I wonder if this is a bug, or if there is a way I can debug why are they not triggered when the PR is opened for the first time.

**Comment by airtower-luna** [comment, marked as answer]
Do you use the `GITHUB_TOKEN` for those actions? In that case [it'll never trigger a workflow run](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow):

> When you use the repository's `GITHUB_TOKEN` to perform tasks, events triggered by the `GITHUB_TOKEN`, with the exception of `workflow_dispatch` and `repository_dispatch`, will not create a new workflow run.

If you want to run workflows from those events, you'll need to provide a PAT.

**Reply by OmarTawfik** [reply]
Thank you! I missed that part of the docs.

> If you want to run workflows from those events, you'll need to provide a PAT.

Wouldn't it be possible to use the same `GITHUB_TOKEN` to send `workflow_dispatch` to start the workflow? For example:

```bash
gh workflow run "check.yml" --ref "my-branch"
```

**Reply by airtower-luna** [reply]
That should be possible, yes. Unless you need the workflow run to provide checks for the PR, AFAIK a `workflow_dispatch` run won't count for that.

**Reply by OmarTawfik** [reply]
Seems like I also need to [manually create/update the check run](https://josh-ops.com/posts/using-github-checks-api/) after I dispatch the workflow. Thanks!
